app/src/lib/stores/app-store.ts

```diff
@@ -3562,14 +3578,14 @@ export class AppStore extends TypedBaseStore<IAppState> {
     })
   }
 
-  private async isCommitting(
+  private async withIsCommitting(
     repository: Repository,
-    fn: () => Promise<string | undefined>
-  ): Promise<boolean | undefined> {
+    fn: () => Promise<boolean>
+  ): Promise<boolean> {
```

Comment:
Aside from renaming this function as you suggested, I also changed the return type to (hopefully) simplify the code, specially after moving things around.